src/monmove.cpp

```diff
@@ -1381,6 +1527,73 @@ bool monster::attack_at( const tripoint &p )
     return false;
 }
 
+bool monster::pickup_at( const tripoint &p, item_location &target )
+{
+    //If called through normal pathfinding, then clear looting effect
+    if( p == goal ) {
+        remove_effect( effect_looting );
+
+    }
+
+    //Item has moved...
+    if( p != target.position() ) {
+        return false;
+    }
+
+    inv.push_back( *target.get_item() );
```

Comment:
```suggestion
    inv.push_back( *target );
```
in addition to what bevapdin said
item_location dereferences into an `item &`